### PR TITLE
Update and edit documentation

### LINKED ISSUE #702: Documentation improvements
Running list of future documentation improvements.

* [x] Cross-reference to usethis (e.g. [`usethis::use_github_links`](http://usethis.r-lib.org/reference/use_github.html)) and the tidyverse style guide (e.g. [writing news bullets](http://style.tidyverse.org/news.html#news-bullets)) to encourage formatting supported by pkgdown.
* [x] Cross-reference specific reference files in `pkgdown.Rmd`. The section on News in the vignette notably lacks several config options.
* [x] Make good use of built-in [highlighting capability](http://pkgdown.r-lib.org/index.html?q=very%20good%20yet#in-the-wild) in those cross-references.
* [x] Also write a section on making good use of said highlighting capability in package documentation
* [x] Use `usethis::use_pkgdown()` in `build_site()` to build-ignore yaml and docs directory. Also document this (#709, #710).
* [x] Once this is all done, delete [the sentence](http://pkgdown.r-lib.org/index.html?q=very%20good%20yet#in-the-wild) that says the documentation isn't any good!

### PR DESCRIPTION
**Pull request by jayhesselberth**
Preview at https://serene-joliot-da0acf.netlify.com/

Closes #702

### CODE REVIEW

File: R/build-reference.R [hunk 1]
```diff
@@ -25,19 +24,27 @@
...
-#' match multiple topics will exclude topics with keyword "internal". To
-#' include, use (e.g.) `starts_with("build_", internal = TRUE)`.
+#' Note that `contents` can contain either a list of function names, or if the
+#' functions in a section share a common prefix or suffix, you can use
+#' `starts_with("prefix")` and `ends_with("suffix")` to select them all. For
+#' more complex naming schemes you can use an aribrary regular expression with
+#' `matches("regexp")`. You can also use a leading `-` to exclude matches from a
+#' section. By default, these functions that match multiple topics will exclude
+#' topics with the Rd keyword "internal". To include these, use (e.g.)
+#' `starts_with("build_", internal = TRUE)`.
```
[the remark this answers is not in the document]
Comment (jayhesselberth) [reply R/build-reference.R]: Not familiar. Link to example?

**Comment by jayhesselberth** [comment]
I would be really nice to have Google Analytics information as well as Docsearch stats from Algolia to be able to know how users interact with documentation and identify areas of expectation and confusion.

Can we set up Google Analytics for pkgdown? I can do this on my side but I think you've done it for other tidyverse packages (?).

I'll reach out to Docsearch to see if we can access pkgdown search statistics via e.g. the Algolia dashboard.

I was thinking it might be nice to create a simple package that does some analysis of GA and Docsearch files from a given site to create a report on changes that could be implemented to improve user experience.

File: R/build-reference.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (hadley) [reply R/build-reference.R]: https://github.com/tidyverse/ggplot2/blob/master/_pkgdown.yml#L24

**Comment by hadley** [comment]
Can you work with @batpigandme to set it up?  It should be possible to get you access to the data too.

**Pull request merged**